apis/core/v1alpha1/resource.go

```diff
@@ -180,12 +185,18 @@ type ClassSpecTemplate struct {
 	// provisioned using this resource class.
 	ProviderReference *corev1.ObjectReference `json:"providerRef"`
 
-	// ReclaimPolicy specifies what will happen to external resources when
-	// managed resources dynamically provisioned using this resource class are
-	// deleted. "Delete" deletes the external resource, while "Retain" (the
-	// default) does not. Note this behaviour is subtly different from other
-	// uses of the ReclaimPolicy concept within the Kubernetes ecosystem per
-	// https://github.com/crossplaneio/crossplane-runtime/issues/21
+	// ReclaimPolicy specifies what will happen to managed resources dynamically
+	// provisioned using this class when their resource claims are deleted, and
+	// what will happen to their underlying external resource when they are
+	// deleted. The "Delete" policy causes the managed resource to be deleted
+	// when its bound resource claim is deleted, and in turn causes the external
+	// resource to be deleted when its managed resource is deleted. The "Retain"
+	// policy causes the managed resource to be retained, in binding phase
+	// "Released", when its resource claim is deleted, and in turn causes the
+	// external resource to be retained when its managed resource is deleted.
+	// The "Retain" policy is used when no policy is specified, however the
```

Comment:
I think we should choose one default for all cases where policy isn't specified and stick with it.

Reply:
I'd prefer to mirror the semantics of the PersistentVolume reclaim policy, which this does.